solidity/ecdsa/contracts/WalletRegistry.sol

```diff
@@ -120,6 +121,15 @@ contract WalletRegistry is
     IStaking public immutable staking;
     IRandomBeacon public randomBeacon;
 
+    /// @notice Allowlist contract for weight-based operator authorization.
+    ///         When set (non-zero address), takes precedence over legacy TokenStaking.
+    ///         This enables gradual migration from T staking to allowlist-based
+    ///         authorization following TIP-092, while maintaining backward
+    ///         compatibility with existing deployments.
+    /// @dev Set via initializeV2() during proxy upgrade. When allowlist is zero
+    ///      address (default), the legacy TokenStaking authorization path is used.
+    Allowlist public allowlist;
```

Comment:
Nice addition for the dual-mode authorization. One thing, since this changes the authorization flow when allowlist is set, have you verified the gas costs dont increase significantly compared to the old TokenStaking path?  The `_currentAuthorizationSource()` indirection adds an extra call on every auth check.